Assignment4/NumberSet/NumSet.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NumberSet
+{
+    public class NumSet
+    {
+        private HashSet<int>? numberSet;

```

Comment:
```suggestion
        private readonly HashSet<int> _NumberSet;
```
No need to mark this as nullable since you will not set it unless you have a non-null value in your constructor. Could also mark as readonly.